hw_lesson13.py

```diff
@@ -0,0 +1,107 @@
+import inspect
+from functools import wraps
+
+"""
+Task 1
+Write a decorator that prints a function with arguments passed to it.
+NOTE! It should print the function, not the result of its execution!
+"""
+
+
+def logger(func):
+    def wrap():
```

Comment:
We are expecting the same function behaviour when decorator being used as without using it.
Try to call your functions with and without decorator providing these test cases:
```
assert square_all(1, 2, 3) == [1, 4, 9]
assert add(2, 3) == 5
```

If it looks a little bit complex, feel free to contact me. :)

Reply:
Fixed it. I ran the test cases, went through the debugger - everything works correctly.